buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/license/RemoteLicenseFetcher.groovy

```diff
@@ -14,155 +14,159 @@
     }
 
     @Override
-    String get() {
-      def doc = Jsoup.connect(BSD_LICENSE_URI.toString()).get()
-
-      TEXT_FORMATTER.getPlainText(doc.select('#content-wrapper')[0])
+    String processDocument(Document doc) {
+      return TEXT_FORMATTER.getPlainText(doc.select("#content-wrapper").get(0))
     }
   }
 
-  static final class CreativeCommonsLicenseFetcher implements RemoteLicenseFetcher {
-    private URI CREATIVE_COMMONS_LICENSE_URI = URI.create("http://creativecommons.org/publicdomain/zero/1.0/")
+  static final class CreativeCommonsLicenseFetcher extends RemoteLicenseFetcher {
 
-    @Override
-    URI getServiceUri() {
-      CREATIVE_COMMONS_LICENSE_URI
+    CreativeCommonsLicenseFetcher() {
+      super("http://creativecommons.org/publicdomain/zero/1.0/")
     }
 
     @Override
-    String get() {
-      def doc = Jsoup.connect(CREATIVE_COMMONS_LICENSE_URI.toString()).get()
-
-      TEXT_FORMATTER.getPlainText(doc.select('#deed'))
+    String processDocument(Document doc) {
+      return TEXT_FORMATTER.getPlainText(doc.select("#deed").get(0))
     }
   }
 
-  static final class MITLicenseFetcher implements RemoteLicenseFetcher {
-    private URI MIT_LICENSE_URI = URI.create("http://www.opensource.org/licenses/mit-license.php")
+  static final class MITLicenseFetcher extends RemoteLicenseFetcher {
 
-    @Override
-    URI getServiceUri() {
-      MIT_LICENSE_URI
+    MITLicenseFetcher() {
+      super("http://www.opensource.org/licenses/mit-license.php")
     }
 
     @Override
-    String get() {
-      def doc = Jsoup.connect(MIT_LICENSE_URI.toString()).get()
-
-      TEXT_FORMATTER.getPlainText(doc.select('#content-wrapper'))
+    String processDocument(Document doc) {
+      return TEXT_FORMATTER.getPlainText(doc.select("#content-wrapper").get(0))
     }
   }
 
-  static final class AnotherMITLicenseFetcher implements RemoteLicenseFetcher {
-    private URI MIT_LICENSE_URI = URI.create("http://opensource.org/licenses/MIT")
+  static final class AnotherMITLicenseFetcher extends RemoteLicenseFetcher {
 
-    @Override
-    URI getServiceUri() {
-      MIT_LICENSE_URI
+    AnotherMITLicenseFetcher() {
+      super("http://opensource.org/licenses/MIT")
     }
 
     @Override
-    String get() {
-      def doc = Jsoup.connect(MIT_LICENSE_URI.toString()).get()
-
-      TEXT_FORMATTER.getPlainText(doc.select('#content-wrapper'))
+    String processDocument(Document doc) {
+      return TEXT_FORMATTER.getPlainText(doc.select("#content-wrapper").get(0))
     }
   }
 
-  static final class GnuClasspathLicenseFetcher implements RemoteLicenseFetcher {
-    private URI GNU_CLASSPATH_LICENSE_URI = URI.create("http://www.gnu.org/software/classpath/license.html")
+  static final class GnuClasspathLicenseFetcher extends RemoteLicenseFetcher {
 
-    @Override
-    URI getServiceUri() {
-      GNU_CLASSPATH_LICENSE_URI
+    // TODO(allisonbm92, vkryachko): This does not fetch the actual license!
```

Comment:
Can you put the missing work in this TODO? Why doesn't it fetch the actual license?

Reply:
Done.